CI failure in unit tests:

```
>       for key in IncludeRole.FROM_ARGS.intersection(my_arg_names):
E       AttributeError: 'tuple' object has no attribute 'intersection'

lib/ansible/playbook/role_include.py:121: AttributeError
```

Integration tests also failed due to the same error.

--- a/lib/ansible/playbook/role_include.py
+++ b/lib/ansible/playbook/role_include.py
@@ -46,7 +46,7 @@ class IncludeRole(TaskInclude):
     BASE = ('name', 'role')  # directly assigned
     FROM_ARGS = ('tasks_from', 'vars_from', 'defaults_from')  # used to populate from dict in role
     OTHER_ARGS = ('private', 'allow_duplicates')  # assigned to matching property
-    VALID_ARGS = list(set(BASE).union(set(FROM_ARGS).union(set(OTHER_ARGS))))  # all valid args
+    VALID_ARGS = tuple(frozenset(BASE + FROM_ARGS + OTHER_ARGS))  # all valid args
 
     # =================================================================================
     # ATTRIBUTES
@@ -113,17 +113,17 @@ class IncludeRole(TaskInclude):
             raise AnsibleParserError("'name' is a required field for include_role.")
 
         # validate bad args, otherwise we silently ignore
-        bad_opts = my_arg_names.difference(set(IncludeRole.VALID_ARGS))
+        bad_opts = my_arg_names.difference(IncludeRole.VALID_ARGS)
         if bad_opts:
             raise AnsibleParserError('Invalid options for include_role: %s' % ','.join(list(bad_opts)))
 
         # build options for role includes
-        for key in set(IncludeRole.FROM_ARGS).intersection(my_arg_names):
+        for key in my_arg_names.intersection(IncludeRole.FROM_ARGS):
             from_key = key.replace('_from', '')
             ir._from_files[from_key] = basename(ir.args.get(key))
 
         # manual list as otherwise the options would set other task parameters we don't want.
-        for option in set(IncludeRole.OTHER_ARGS).intersection(my_arg_names):
+        for option in my_arg_names.intersection(IncludeRole.OTHER_ARGS):
             setattr(ir, option, ir.args.get(option))
 
         return ir

--- a/lib/ansible/playbook/role_include.py
+++ b/lib/ansible/playbook/role_include.py
@@ -118,12 +118,12 @@ class IncludeRole(TaskInclude):
             raise AnsibleParserError('Invalid options for include_role: %s' % ','.join(list(bad_opts)))
 
         # build options for role includes
-        for key in IncludeRole.FROM_ARGS.intersection(my_arg_names):
+        for key in set(IncludeRole.FROM_ARGS).intersection(my_arg_names):
             from_key = key.replace('_from', '')
             ir._from_files[from_key] = basename(ir.args.get(key))
 
         # manual list as otherwise the options would set other task parameters we don't want.
-        for option in IncludeRole.OTHER_ARGS.intersection(my_arg_names):
+        for option in set(IncludeRole.OTHER_ARGS).intersection(my_arg_names):
             setattr(ir, option, ir.args.get(option))
 
         return ir

